### PR TITLE
8287132: Retire Runtime.runFinalizersOnExit so that it always throws UOE

### PR DESCRIPTION
**Pull request by dholmes-ora**
Forward port of the changes for SE 8 Maintenance release 4, as applied to jdk8u-ri, for:

[JDK-8287132](https://bugs.openjdk.org/browse/JDK-8287132): Retire Runtime.runFinalizersOnExit so that it always throws UOE

The changes applied "unclean" only with regard to trivial things like copyright year updates.

Thanks.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8287132](https://bugs.openjdk.org/browse/JDK-8287132): Retire Runtime.runFinalizersOnExit so that it always throws UOE

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u pull/13/head:pull/13` \
`$ git checkout pull/13`

Update a local copy of the PR: \
`$ git checkout pull/13` \
`$ git pull https://git.openjdk.org/jdk8u pull/13/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13`

View PR using the GUI difftool: \
`$ git pr show -t 13`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u/pull/13.diff">https://git.openjdk.org/jdk8u/pull/13.diff</a>

</details>

### CODE REVIEW

**Review by irisclark — APPROVED** [review]
Spec change match JSR 337 MR 4, as expected.

I leave it to the 8u Maintainers to comment on whether these changes should be applied to the jdk8u repo or the jdk8u-dev repo.

**Comment by dholmes-ora** [comment]
This is the wrong repo so I will close and re-submit.

**Comment by dholmes-ora** [comment]
Resubmitted as https://github.com/openjdk/jdk8u-dev/pull/92

**Pull request closed without merging**
